networkx/algorithms/approximation/broadcasting.py

```diff
@@ -0,0 +1,139 @@
+import networkx as nx
+from networkx import NetworkXError
+from networkx.utils import not_implemented_for
+
+__all__ = [
+    "tree_broadcast_center",
+    "tree_broadcast_time",
+]
+
+
+def _get_max_broadcast_value(G, U, v, values):
+    adj = sorted(set(G.neighbors(v)) & U, key=lambda u: values[u], reverse=True)
+    return max(values[u] + i for i, u in enumerate(adj, start=1))
+
+
+def _get_broadcast_centers(G, v, values, target):
+    adj = sorted(set(G.neighbors(v)), key=lambda u: values[u], reverse=True)
+    j = min(i for i, u in enumerate(adj, start=1) if values[u] + i == target)
```

Comment:
This is the helper function for getting all the broadcast centers from the last vertex in the algorithm. Please let me know if you have any questions.

Reply:
also, is there a better way to get $j$ in this scenario? I could return the first index where ``values[u]+ i == target`` to avoid calculating the entire thing. Here is the pseudocode for the algorithm below (we have already calculated $max \{ ... \} $ from ``_get_max_broadcast_value``):
<img width="887" alt="image" src="https://github.com/networkx/networkx/assets/89562186/15c74408-2d3f-4f60-b9d1-d83f0061ffb4">